torch/nn/init.py

```diff
@@ -525,3 +527,104 @@ def deprecated_init(*args, **kwargs):
 kaiming_normal = _make_deprecate(kaiming_normal_)
 orthogonal = _make_deprecate(orthogonal_)
 sparse = _make_deprecate(sparse_)
+
+
+v = torch.__version__.split('.')
+# These can also be used as global flags
+_torch_version = (int(v[0]), int(v[1]), int(v[2][0]))
```

Comment:
We'll have to be careful that there's not ever a version `x.y.12` since will `nt(v[2][0])` mishandle that. probably safe?

Reply:
Thanks, @elistevens . I think it would be best to use the same re pattern as `torch.utils.cpp_extension.BUILT_FROM_SOURCE_VERSION_PATTERN`. That way we have an even number of breakages, hopefully `0`.

Reply:
Changed the `_torch_version` logic with re search.

```python
def _get_torch_version():
    """Returns torch.__version__ as Tuple(int,int,int)"""
    GET_TORCH_VERSION = re.compile(r'\d+\.\d+\.\d+')
    v = GET_TORCH_VERSION.findall(torch.__version__)
    assert len(v) == 1, "GET_TORCH_VERSION failed to find torch version"
    v = v[0].split('.')
    return (int(v[0]), int(v[1]), int(v[2]))
```